active/0000-engines.md

```diff
@@ -0,0 +1,411 @@
+});
+```
+
+> Note: If unspecified, `path` will match the name of the engine.
+
+Calls to `mount` can be nested within routes. An engine can be mounted at
+multiple routes, and each will represent a new instance of the engine to be
+created.
+
+### Mounting route-less engines
+
+A `mount()` DSL will also be added to routes, which will enable embedding of
+route-less engines in outlets. This can be called from `renderTemplate` (or
+`renderComponents` once routable components are introduced).
+
+`mount` has a similar signature to `render`, although it is obviously
+engine-specific instead of template-specific. `mount` can be used to specify
+a target template and outlet as follows:
+
+```
+renderTemplate: function() {
+  // Mount the chat engine in the sidebar
+  this.mount('chat', {
+    into: 'main',
+    outlet: 'sidebar'
+  });
+}
+```
+
+As a result, the engine's `application` template will be rendered into the
+`sidebar` outlet in the application's `main` template.
+
+### Loading phases
+
+Engines can exist in several phases:
+
+* Booted - an engine that's been installed in a parent application will have
+  its dependencies loaded and its (non-instance) initializers invoked when the
+  parent application boots.
+
+* Mounted - Routable and route-less engines have slightly different concepts of
+  "mounting". A routable engine is considered mounted when it has been included
+  by a router at one or more mount-points. A route-less engine is considered
+  mounted as soon as a route's `mount` call resolves.
+
+* Instantiated - When an engine is instantiated, an `EngineInstance` is created
+  and an engine's instance initializers are invoked. A routable engine is
+  instantiated when a route is visited at or beyond its mount-point. A
+  route-less engine is instantiated as soon as it is mounted.
+
+Special `before` and `after` hooks could be added to application instance
+initializers that allow them to be ordered relative to engine instance
+initializers.
+
+### Engine boundaries
+
+Besides its routing map, an engine does not share any other resources with its
+parent by default. Engines maintain their own registries and containers, which
+ensure that they stay isolated. However, some explicit sharing of resources
+between engines and parents is allowed.
+
+#### Engine / parent dependencies
+
+Dependencies between engines and parents can be defined imperatively or
+declaratively.
+
+Imperative dependencies can be defined in an engine's instance initializers.
+When an engine is instantiated, the `parent` property on its `EngineInstance` is
+set to its parent instance (either an `ApplicationInstance` or
+`EngineInstance`). Since the engine instance is available in the instance
+initializer, this `parent` property can also be accessed. This allows an engine
+instance to interrogate its parent, specifically through its `RegistryProxy` and
+`ContainerProxy` interfaces.
+
+Alternatively, declarative dependencies can be defined on a limited basis. The
```

Comment:
This seems like it will become extremely fragile, however it is unclear to me how else to deal with Ember's runtime dependencies.